src/sync/utils/create-build-array-actions.js

```diff
@@ -0,0 +1,107 @@
+const REGEX_NUMBER = new RegExp(/^\d+$/)
+const REGEX_UNDERSCORE_NUMBER = new RegExp(/^_\d+$/)
+
+export const ADD_ACTIONS = 'create'
+export const REMOVE_ACTIONS = 'remove'
+export const CHANGE_ACTIONS = 'change'
+
+/**
+ * Generate + configure a function to build actions for nested objects
+ * @param  {string} key    key of the attribute containing the array of
+ *   nested objects
+ * @param  {object} config configuration object that can contain the keys
+ *   [ADD_ACTIONS, REMOVE_ACTIONS, CHANGE_ACTIONS], each of
+ *   which is a function. The function should accept the old + new arrays and
+ *   return an action object.
+ * @return {Array}        The generated array of actions
+ */
+export default function createBuildArrayActions (key, config) {
+  return function buildNestedObjectActions (diff, oldObj, newObj) {
+    const addActions = []
+    const removeActions = []
+    const changeActions = []
+
+    if (diff[key]) {
+      const nestedObjects = diff[key]
+      Object.keys(nestedObjects).forEach((index) => {
+        if (config[ADD_ACTIONS] && isCreateAction(nestedObjects, index))
+          addActions.push(
+            config[ADD_ACTIONS](oldObj[key], newObj[key], index)
```

Comment:
It is simpler to pass the same arguments to those functions, on the other hand it is simpler from who's using the function to get only the useful values.